packages/app/src/app/pages/SignIn/ChooseProvider.tsx

```diff
@@ -0,0 +1,174 @@
+import React from 'react';
+import { useAppState, useActions } from 'app/overmind';
+import {
+  AppleIcon,
+  github as GitHubIcon,
+  CodeSandboxIcon,
+  GoogleIcon,
+} from '@codesandbox/components/lib/components/Icon/icons';
+import {
+  Element,
+  Stack,
+  Text,
+  Button as MainButton,
+} from '@codesandbox/components';
+import { css } from '@styled-system/css';
+import history from 'app/utils/history';
+
+import { Button } from './components/Button';
+
+interface ChooseProviderProps {
+  redirectTo?: string;
+  onSignIn?: () => void;
+}
+
+export const ChooseProvider = ({
```

Comment:
This component used to be `packages/app/src/app/pages/SignIn/Modal.tsx`. Most logic is copied from there, most significant changes is the `handleSignIn` function, which used to be three different functions.